vignettes/npi.Rmd

```diff
@@ -0,0 +1,70 @@
+---
+title: "Introduction to npi"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to npi}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(npi)
+``` 
+
+This vignette provides an brief introduction to the npi package. 
+
+`npi` is an R package that allows R users to access the [U.S. National Provider Identifier (NPI) Registry](https://npiregistry.cms.hhs.gov/) API by the Center for Medicare and Medicaid Services (CMS). 
+
+The package makes it easy to obtain administrative data linked to a specific individual or organisational healthcare provider. Additionally, users can perform advanced searches based on provider name, location, type of service, credentials, and many other attributes.
+
+## Search registry
+
+To explore organizational providers with primary locations in New York City, we could use the `city` argument in the `npi_search()`. The nyc dataset here finds 10 organizational providers with primary locations in New York City, since 10 is the default number of records that are return in `npi_search()`. The response is a tibble that has high-cardinality data organized into list columns.
+
+```{r}
+nyc <- npi_search(city = "New York City")
+nyc
+```
+
+Other search arguments for the function include npi number, enumeration type, taxonomy description, individual provider's first name, individual provider's last name, use first name alias, organization name, address purpose, state, postal code, country code, and limit.
+
+Visit the function's help page via `?npi_search` after installing and loading the package for more details.
+
+## Increasing number of records returned
+
+The `limit` argument lets you set the maximum records to return from 1 to 1200 inclusive. The default of 10.
+
+Because the API returns up to 200 records per request, values of limit greater than 200 will result in multiple API calls.
+
+```{r}
+nyc_25 <- npi_search(city = "New York City", limit = 25)
```

Comment:
Can we run this with a limit value greater than 200 to illustrate the auto-paging feature, or will that create problems in a vignette?

Reply:
Looks like we can without any problems. The print is automatically shortened. Done.